libraries/dotnet/EnergyOriginAuthorization/EnergyOriginAuthorization.Tests/HttpClientExtensionTests.cs

```diff
@@ -0,0 +1,22 @@
+using System.Net.Http;
+using EnergyOriginAuthorization;
+using Xunit;
+
+namespace Tests;
+
+public sealed class HttpClientExtensionTests
+{
+    [Fact]
+    public void AddAuthorizationTokenTest_valid_token()
+    {
+        var client = new HttpClient();
+        var context = new AuthorizationContext("subject", "actor", "token");
+
+        client.AddAuthorizationToken(context);
+
+        Assert.NotNull(client);
+        Assert.NotEmpty(client.DefaultRequestHeaders);
+        Assert.Equal(context.Token, client.DefaultRequestHeaders.Authorization?.Parameter ?? "");
+        Assert.Equal("Bearer", client.DefaultRequestHeaders.Authorization?.Scheme ?? "");

```

Comment:
Why are these allowed nullable types?
As in token, shouldn't there always be some kind of token, or else get denied in-front?

Reply:
@Exobitt The nullable type is from `HttpClient` in `System.Net.Http`.